From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Tue, 25 Apr 2023 04:53:22 -0400
Subject: [PATCH] require HTTPS for component updates

Adapted from GrapheneOS Vanadium patch 0068-require-HTTPS-for-component-updates.patch for the Vandium browser port.

---
 .../component_updater/aw_component_updater_configurator.cc      | 2 +-
 .../component_updater/chrome_component_updater_configurator.cc  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/chrome/browser/component_updater/chrome_component_updater_configurator.cc b/chrome/browser/component_updater/chrome_component_updater_configurator.cc
index 2994ebe96b9d1..a69500dd91397 100644
--- a/chrome/browser/component_updater/chrome_component_updater_configurator.cc
+++ b/chrome/browser/component_updater/chrome_component_updater_configurator.cc
@@ -116,7 +116,7 @@ class ChromeConfigurator : public update_client::Configurator {
 ChromeConfigurator::ChromeConfigurator(const base::CommandLine* cmdline,
                                        PrefService* pref_service)
     : configurator_impl_(ComponentUpdaterCommandLineConfigPolicy(cmdline),
-                         /*require_encryption=*/false),
+                         /*require_encryption=*/true),
       pref_service_(pref_service),
       persisted_data_(update_client::CreatePersistedData(
           base::BindRepeating(
